staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -257,7 +260,7 @@ func (rules *ClientConfigLoadingRules) Migrate() error {
 		return nil
 	}
 
-	for destination, source := range rules.MigrationRules {
+	for source, destination := range rules.MigrationRules {
```

Comment:
The method of migrating is problematic, because it copies a file (which could be actively managed and contain credentials or CA bundles that need periodic updating) to another location which is not actively managed. This detaches the active configuration file from the location that was being updated and sets up a time-bomb when the credentials or certificate bundles expire.

Reply:
Understood. This change would require documentation and spread the world via blog posts etc. Not GA until 'X' releases.

Reply:
Going to update the KEP as requested.